packages/models/src/token-generation-readmodel/token-generation-states-entry.ts

```diff
@@ -0,0 +1,55 @@
+import { z } from "zod";
+import { ClientKind } from "../authorization/client.js";
+import {
+  AgreementId,
+  ClientId,
+  PurposeId,
+  PurposeVersionId,
+  TenantId,
+} from "../brandedIds.js";
+import { ItemState } from "./platform-states-entry.js";
+import {
+  TokenGenerationStatesClientKidPurposePK,
+  TokenGenerationStatesClientKidPK,
+  GSIPKConsumerIdEServiceId,
+  GSIPKEServiceIdDescriptorId,
+  GSIPKClientIdPurposeId,
+} from "./dynamoDB-keys.js";
+
+const TokenGenerationStatesBaseEntry = z.object({
+  consumerId: TenantId,
+  clientKind: ClientKind,
+  publicKey: z.string(),
+  GSIPK_clientId: ClientId,
+  GSIPK_kid: z.string(),
+  GSIPK_clientId_purposeId: GSIPKClientIdPurposeId,
+  updatedAt: z.string().datetime(),
+});
+type TokenGenerationStatesBaseEntry = z.infer<
+  typeof TokenGenerationStatesBaseEntry
+>;
+
+export const TokenGenerationStatesClientPurposeEntry =
+  TokenGenerationStatesBaseEntry.extend({
```

Comment:
in order to handle error cases described [here](https://pagopa.atlassian.net/wiki/spaces/PDNDI/pages/1203208195/Propagazione+stati+piattaforma#Caso-2), attributes related to purpose, agreement and eservice should be optional

Reply:
Updated